docs/3.tutorials/auction/5-frontend.md

```diff
@@ -0,0 +1,220 @@
+
+<Language value="javascript" language="javascript" showSingleFName={true}>
+    <Github fname="index.js"
+        url="https://github.com/near-examples/auctions-tutorial/blob/main/frontend/src/pages/index.js#L54-L70"
+        start="54" end="70" />
+</Language>
+
+Note that this effect will only run once the `auctionInfo` updates because we first need the NFT contract ID and token ID from `auctionInfo` to make a valid call to `nft_token`.
+
+In the `AuctionItem` component we display the NFT image, name, and description.
+
+<Language value="javascript" language="javascript" showSingleFName={true}>
+<Github fname="AuctionItem.jsx"
+    url="https://github.com/near-examples/auctions-tutorial/blob/main/frontend/src/components/AuctionItem.jsx"
+    />
+</Language>
+
+Note that an image caching service is used to display the NFT image for better performance.
+
+---
+
+## Making a bid
+
+To make a bid we call the `ft_transfer_call` method on the FT contract which subsequently calls `ft_on_transfer` on the auction contract and attaches fungible tokens to the call.
+
+<Language value="javascript" language="javascript">
+    <Github fname="index.js" 
+        url="https://github.com/near-examples/auctions-tutorial/blob/main/frontend/src/pages/index.js#L99-L109"
+        start="99" end="109" />
+    <Github fname="near.js" 
+        url="https://github.com/near-examples/auctions-tutorial/blob/main/frontend/src/wallets/near.js#L107-L126"
+        start="107" end="126"/>
+</Language>
+
+We now multiply the bid amount by the decimals of the FT to get the correct amount to send. Since this method requires a 1 yoctoNEAR deposit the wallet will prompt the user to sign the transaction.
+
+---
+
+## Claiming the auction
+
+Once the auction is over (the current time is greater than the end time) the auction can be claimed. At this point, the timer will be hidden and a button to claim the auction will be displayed. Once clicked the `claim` method will be called on the auction contract to send the highest bidder the NFT and the auctioneer the FTs.
+
+<Language value="javascript" language="javascript" showSingleFName={true}>
+    <Github fname="index.js" 
+        url="https://github.com/near-examples/auctions-tutorial/blob/main/frontend/src/pages/index.js#L111-L118"
+        start="111" end="118" />
+</Language>
+
+---
+
+## Conclusion
+
+In this part of the tutorial, we have implemented a simple frontend for a NEAR contract. Along the way we have learned how to use the wallet selector to sign the user in and out, how to view the contract’s state, how to sign and send transactions, and use ft_transfer_call from a frontend.
+
+Whilst we can see the highest bid we may want to see the auction's bidding history. Since the contract only stores the most recent bid we need to use an indexer to pull historical data. In the [next part](./6-indexing.md) of the tutorial, we'll look at quering historical data using an API endpoint.
```

Comment:
```suggestion
While we can see the highest bid, we may want to see the auction's bidding history. Since the contract only stores the most recent bid, we need to use an indexer to pull historical data. In the [next part](./6-indexing.md) of the tutorial, we'll look at querying historical data using an API endpoint.
```